src/odemis/gui/comp/miccanvas.py

```diff
@@ -656,6 +656,8 @@ def fit_to_bbox(self, bbox):
           but with explicit bounding box, so we can zoom to a specific position.
         bbox (4 floats): bounding box to be shown, defined as minx, miny, maxx,
           maxy positions in m.
+
+        Note: Should be called from main GUI thread.
```

Comment:
If it is already called from the main GUI thread I would update this comment to: `Note: Is called from main GUI thread`. If it should be forced to be called from the main GUI thread, I think you need to add the `@call_in_wx_main` decorator to this function.

Reply:
I added this comment in concordance with Éric's review. The "should" indicates that every caller should already be running in the main GUI thread, so also potential future callers. I think that is a bit the difference to "Is" - this does not highlight the need that I need to make sure the caller is already running in the main GUI thread. 
I can extend the comment and add "Caller should run in main GUI thread".

I don't know if the decorator makes sense here. The part that changes the GUI is the drawing update, which is in a wx.CallAfter call. From what I understand this makes sure that this call is only executed after the current and pending event handlers have been completed. But I don't understand why we still need to make sure that the caller of this method runs in the main gui thread. @pieleric can you clarify?

Reply:
Then can you maybe add what call makes sure that this is called from the main GUI thread?